pkg/application/stateDiffUseCase.go

```diff
@@ -45,19 +59,15 @@ func (useCase *StateDiffUseCase) DetermineStateDiff(ctx context.Context, bluepri
 		return fmt.Errorf("cannot load blueprint spec %q to determine state diff: %w", blueprintId, err)
 	}
 
-	logger.Info("determine state diff to the cloudogu ecosystem", "blueprintStatus", blueprintSpec.Status)
-	installedDogus, err := useCase.doguInstallationRepo.GetAll(ctx)
-	if err != nil {
-		return fmt.Errorf("cannot get installed dogus to determine state diff: %w", err)
-	}
-
-	installedComponents, err := useCase.componentInstallationRepo.GetAll(ctx)
+	logger.Info("collect ecosystem state for state diff")
+	ecosystemState, err := useCase.collectEcosystemState(ctx, blueprintSpec.EffectiveBlueprint)
 	if err != nil {
-		return fmt.Errorf("cannot get installed components to determine state diff: %w", err)
+		return fmt.Errorf("could not determine state diff: %w", err)
```

Comment:
Either i would use a precise error message like `could not collect ecosystem state` or just return the error.

Reply:
`could not collect ecosystem state` is the exact message which comes from the `collectEcosystemState` function. My thought here was to add the information that we used this to determine the state diff.

Resulting errors would like like this:

`could not determine state diff: could not collect ecosystem state: could not connect to etcd`
or
`could not determine state diff: could not decrypt sensitive dogu config: no private key found for ...`

Is that ok in this way or do you think we should change that nonetheless?

Reply:
Okay since the caller just returns the plain error this is ok :)